app/uk/gov/hmrc/helptosavefrontend/util/lock/Lock.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.util.lock
+
+import akka.actor.{Actor, Cancellable, Props, Scheduler}
+import akka.pattern.pipe
+import org.joda.time
+import reactivemongo.api.DB
+import uk.gov.hmrc.helptosavefrontend.util.Logging
+import uk.gov.hmrc.lock.{ExclusiveTimePeriodLock, LockMongoRepository, LockRepository}
+import uk.gov.hmrc.helptosavefrontend.util.lock.LockProvider.ExclusiveTimePeriodLockProvider
+import uk.gov.hmrc.helptosavefrontend.util.toFuture
+
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success}
+
+/**
+ * This is an `Actor` which handles changing of state when a lock is acquired or a lock
```

Comment:
Please can you add ... The objective of the Lock is to provide a single instance of a resource that is used in a multi-instance microservice environment.

....also, to mention what property name controls the period time with an example setting

Reply:
That comment doesn't belong in the scaladoc for `Lock` since that actor uses the `LockProvider` trait which is generic and does not know anything about mongo lock. I've added documentation in `ExclusiveTimePeriodLockProvider` though to make it clearer what the intention is

Reply:
For the `Lock` actor the lock duration is defined by the `LockProvider` being passed in - I've made this clear in the documentation now. The `Lock` actor does not know anything about how the lock duration is defined in the `LockProvider` since `LockProvider` is generic